pkg/ppm/bounds.go

```diff
@@ -70,6 +75,23 @@ func parseBoundAsDateTime(partition postgresql.PartitionResult) (lowerBound, upp
 	return lowerBound, upperBound, nil
 }
 
+func parseBoundAsDateTimeWithTimezone(partition postgresql.PartitionResult) (lowerBound, upperBound time.Time, err error) {
+	lowerBound, err = time.Parse("2006-01-02 15:04:05Z07", partition.LowerBound)
+	if err != nil {
+		return time.Time{}, time.Time{}, fmt.Errorf("can't parse lowerbound as datetime with timezone: %w", err)
+	}
+
+	upperBound, err = time.Parse("2006-01-02 15:04:05Z07", partition.UpperBound)
+	if err != nil {
+		return time.Time{}, time.Time{}, fmt.Errorf("can't parse upperbound as datetime with timezone: %w", err)
+	}
+
+	lowerBound = convertToDateTimeWithoutTimezone(lowerBound)
+	upperBound = convertToDateTimeWithoutTimezone(upperBound)
```

Comment:
Just FYI, what you're suggesting will fix the parsing issue but as I mentioned in your previous MR about datetime with timezone, I'm worried losing the timezone information could lead to akward bugs down the line if we happen to run PPM in different timezones.
I guess for our current use case we can let it as-is but I prefer us to be aware of it.

Reply:
Hey @dcupif , 
Indeed, the PPM is currently using `time.Now()` (UTC) while provisioning and checking partitions.  When the DB timezone is different and UTC time and local time don't fall into the same day (eg, `2024-10-31T23:00:00-01:00`), if we run PPM, we will use the wrong bounds for the provisioning. The partition check will also fail.

I think it would be better to distinguish the DateTime w/ timezone and DateTime w/o timezone and the PPM will always use UTC time to compute the bounds and it preserves the timezone information while formatting a bound if the data type is DateTime w/ timezone.

To distinguish 2 DateTimes, we could introduce a new ColumnType [here](https://github.com/qonto/postgresql-partition-manager/blob/main/internal/infra/postgresql/column.go#L40-L41)
```golang
	case "timestamp without time zone":
-		return DateTime, nil
+		return DateTimeWithTimezone, nil
```

To preserve the timezone, we could convert the bound with timezone to UTC, then format it to a date.
```golang
func convertToDateTimeWithoutTimezone(bound time.Time) time.Time {
- 	parsedTime, err := time.Parse("2006-01-02 15:04:05", bound.Format("2006-01-02 15:04:05"))
+	parsedTime, err := time.Parse("2006-01-02 15:04:05", bound.UTC().Format("2006-01-02 15:04:05"))
	if err != nil {
		return time.Time{}
	}
	return parsedTime
}
```

Wdyt?

Reply:
I opened another PR https://github.com/qonto/postgresql-partition-manager/pull/52 to keep the timezone information while fetching the partition bounds from postgresql.

Reply:
I'm going to resolve this comment. Please let me know what do you think about the above PR.

Reply:
I've approved the other PR.